src/pages/studyView/table/treatments/AbstractTreatmentsTable.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { action, computed, observable } from 'mobx';
+import autobind from 'autobind-decorator';
+import {
+    PatientTreatmentRow,
+    SampleTreatmentRow,
+} from 'cbioportal-ts-api-client';
+import { SelectionOperatorEnum } from 'pages/studyView/TableUtils';
+import styles from 'pages/studyView/table/tables.module.scss';
+import { treatmentUniqueKey, TreatmentTableType } from './treatmentsTableUtil';
+import { stringListToSet } from 'cbioportal-frontend-commons';
+import { SortDirection } from 'shared/components/lazyMobXTable/LazyMobXTable';
+
+type TreatmentsTableProps = {
+    filters: string[][];
+    tableType: TreatmentTableType;
+    onUserSelection: (value: string[][]) => void;
+};
+
+export abstract class TreatmentsTable<
+    P extends TreatmentsTableProps,
+    S
```

Comment:
you need to parametrize the react state S?

Reply:
Good call.